### DISCUSSION TITLE
BlenderBot2 - its answer to a question for latest news

### DISCUSSION THREAD

**Discussion by linlinsam** [facebookresearch/ParlAI#3904] [Q&A]
Hello, I am testing Blenderbot2 and expected to get right answers to my question for latest news. But I got the following answers.

Enter Your Message: Who is the US president now?
[BlenderBot2Fid]: I am not sure but I think it is Donald Trump, he is the current president of the United States.
Enter Your Message: Where is the current summer Olympic being held?
[BlenderBot2Fid]: The current summer olympics are being held in Rio de Janeiro, Brazil. Have you ever been there?

I tried the both and got similar responses. Both referred to the previous information.
- parlai interactive -mf zoo:blenderbot2/blenderbot2_400M/model --search-server relevant_search_server
- parlai interactive -mf zoo:blenderbot2/blenderbot2_400M/model --search-server "https://www.google.com"

Did I do something wrong? Should I provide another value for "--search-server" argument?

Thank you.

**Comment by superchargez** [comment]
Same thing occoured when I tested. It thinks that it is 2017 still. And it can't tell date or time or day of the week either. Also, it still does not know that GPT-3 was out. Could this problem be solved if we use a search_service other than 0.0.0.0:8080? What would be the best case for this? Google?

**Comment by klshuster** [comment]
BB2 currently uses a model to determine whether to access long-term memory (from the conversation) or the internet. it's possible the model is not searching the internet for these questions and thus is answering incorrectly.

If you'd like to force the model to search the internet, try setting the `--query-generator-model-file` to `zoo:sea/bart_sq_gen/model`

**Reply by superchargez** [reply]
Thank you Kurt I tried but colab crashed due to insufficient memory. I think model is not configured correctly. I mean if it needs to look for information online should not it consume less RAM? compared to if it has to look for something in internal storage/database or the model itself (parameters/nodes creating memory)?

It crashed so I don't know how much RAM does it require to run smoothly. Other than that model runs ok in colab with Tesla T4 (16GB, VRAM, roughly uses 11GB) and 12 GM RAM (it uses roughly 4GB)

**Reply by klshuster** [reply]
searching the internet, and conditioning on retrieved documents, would require more RAM, as the model must encode information across several documents (as opposed to just the dialogue history)

**Comment by linlinsam** [comment]
@klshuster: Thank you for the answer. But I still have an issue.

I tried this:
`parlai interactive -mf zoo:blenderbot2/blenderbot2_400M/model --search-server relevant_search_server --query-generator-model-file zoo:sea/bart_sq_gen/model
`
Then, got the following error: (Please see the attached text file to see the full message.)
_Failed to establish a new connection: [Errno 11001] getaddrinfo failed'))_
[Errno 11001-getaddrinfo failed.txt](https://github.com/facebookresearch/ParlAI/files/6942204/Errno.11001-getaddrinfo.failed.txt)

Then, I changed the --search-server to google as below:
`parlai interactive -mf zoo:blenderbot2/blenderbot2_400M/model --search-server "https://www.google.com" --query-generator-model-file zoo:sea/bart_sq_gen/model`

The following is the result:
Enter Your Message: Hi, do you know who the US president is now?
08:41:34 | Failed to retrieve data from server! Search server returned status 405
08:41:34 | Server search did not produce any results for "Donald Trump" query. returning an empty set of results for this query.
[BlenderBot2Fid]: No, I don't know who that is. _POTENTIALLY_UNSAFE__
Enter Your Message: Where is the Olympics being held now?
08:42:50 | Failed to retrieve data from server! Search server returned status 405
08:42:50 | Server search did not produce any results for "Olympics 2021" query. returning an empty set of results for this query.
[BlenderBot2Fid]: I'm not sure, but I do know that the Olympics are held in Rio de Janeiro, Brazil.

Should I provide something else to the --search-server argument? Thank you.

**Reply by klshuster** [reply]
you will need to configure an appropriate search server for the agent. please search through the parlai issues for more information

**Reply by linlinsam** [reply]
Okay, thank you.

**Comment by Adrian-1234** [comment]
A quick Blender 2.0 test, with and without the --query-generator-model-file switch:
(I ran the two chats concurrently)

Note: Whether the --query-generator-model-file switch is used or not,  the search server still appears to make queries.

Applying the --query-generator-model-file switch  doesn't appear to be making things much better:

------------------------------------------------------------------

Firstly, started search server with:
python ./ParlAI_SearchEngine/search_server.py serve --host 0.0.0.0:8080

------------------------------------------------------------------

parlai interactive -mf zoo:blenderbot2/blenderbot2_3B/model --search-server  http://0.0.0.0:8080
Enter Your Message: Tell me what is in the news regarding Afghanistan  and the Taliban ?

Enter Your Message: What does al jazeera say about the Taliban in Afghanistan  today ?
[BlenderBot2Fid]: I'm not sure what you're asking. _POTENTIALLY_UNSAFE__

Enter Your Message: Is Kabul airport likely to be under attack?
[BlenderBot2Fid]: No, I think it will be fine. I think they're just reporting it because it's newsworthy.

Enter Your Message: What is happening in Afghanistan today?
It's the anniversary of the fall of the Soviet Union, so a lot of people are talking about it.

Enter Your Message: What did Joe Biden said about America's withdrawal from Afghanistan ?
I'm not sure what he said about it, but it was a big deal when it happened.

Enter Your Message: Following the US drone strike in Afghanistan today, did any civilian die?
[BlenderBot2Fid]: Not that I know of, but I haven't been able to keep up with the news lately.

-------------------------------------------------------------------------

Now with the --query-generator-model-file switch:

parlai interactive -mf zoo:blenderbot2/blenderbot2_3B/model --search-server http://0.0.0.0:8080 --query-generator-model-file zoo:sea/bart_sq_gen/model

Enter Your Message: Tell me what is in the news regarding Afghanistan and the taliban ?
[BlenderBot2Fid]: I am not sure what is going on with the taliban. I do know that they have been fighting for a long time.


Enter Your Message: What does al jazeera say about the taliban in Afghanistan today ?
[BlenderBot2Fid]: I do not know what they have to say about it. Do you have any information about it?

Enter Your Message: Is kabul airport likely to be under attack?
Enter Your Message: [BlenderBot2Fid]: No, I don't think so. I think it will just be a normal day at the airport.

Enter Your Message: What is happening in Afghanistan today?
Enter Your Message: [BlenderBot2Fid]: I'm not sure. I haven't been able to keep up with the news lately. What about you?

Enter Your Message: What did Joe Biden said about America's withdrawal from Afghanistan ?
Enter Your Message: [BlenderBot2Fid]: He said that it was a mistake to go in the first place and that we should have stayed out.

Enter Your Message: Following the US drone strike in Afghanistan today, did any civilian die?
[BlenderBot2Fid]: I'm not sure. I didn't see anything about it on the news. I hope not.


--------------------------------------------------------------------------------------------

Here is an example tail of the web search server log:

 > Result: Title: `Reports children killed in US attack on suicide bombers ?? as it happened | World news | The Guardian`
   https://www.theguardian.com/world/live/2021/aug/29/afghanistan-live-news-terror-attack-highly-likely-in-next-24-36-ho
urs-says-biden-last-uk-troops-leave-kabul
 > Result: Title: `??Biden??s worst nightmare??: how Afghanistan shook a president | Joe Biden | The Guardian`
   https://www.theguardian.com/us-news/2021/aug/28/joe-biden-worst-nightmare-afghanistan-islamic-state-bomb-marines-leon
-panetta
 > Result: Title: `Republicans scent blood as Biden assailed over Afghanistan pullout | Joe Biden | The Guardian`
   https://www.theguardian.com/us-news/2021/aug/29/joe-biden-afghanistan-republicans-midterms
127.0.0.1 - - [30/Aug/2021 11:02:41] "POST / HTTP/1.1" 200 -
news | The Guardian`
   https://www.theguardian.com/world/live/2021/aug/30/afghanistan-live-news-kabul-taliban-latest-updates-airport-attack-
islamic-state-isis-k-us-drone-strike-civilians-children-killed-deaths-reported
 > Result: Title: `Reports children killed in US attack on suicide bombers ?? as it happened | World news | The Guardian`
   https://www.theguardian.com/world/live/2021/aug/29/afghanistan-live-news-terror-attack-highly-likely-in-next-24-36-ho
urs-says-biden-last-uk-troops-leave-kabul
 > Result: Title: `Afghanistan: Children among civilians killed in US drone attack | ISIL/ISIS News | Al Jazeera`
   https://www.aljazeera.com/news/2021/8/30/afghanistan-children-among-civilians-killed-in-us-drone-strike
127.0.0.1 - - [30/Aug/2021 14:52:30] "POST / HTTP/1.1" 200 -
